packages/@react-stately/grid/src/GridCollection.ts

```diff
@@ -23,92 +23,106 @@ export class GridCollection<T> implements IGridCollection<T> {
   keyMap: Map<Key, GridNode<T>> = new Map();
   columnCount: number;
   rows: GridNode<T>[];
+  sections: GridNode<T>[];
 
   constructor(opts?: GridCollectionOptions<T>) {
     this.keyMap = new Map();
     this.columnCount = opts?.columnCount;
     this.rows = [];
+    this.sections = [];
+
+    let visit = (node: GridNode<T>, i?: number) => {
```

Comment:
Refactored GridCollection a bit via [6f60708](https://github.com/adobe/react-spectrum/pull/4210/commits/6f607084ff8ed1ec638ce72fc8679d81d75e724f), see that commit and [ca029a9](https://github.com/adobe/react-spectrum/pull/4210/commits/ca029a9e517876169b23042faf416492a48fda53) for a bit of background. 6f60708 might be a bit excessive perhaps given how we'll be eventually converting to the new way of doing Collections from RAC, but it felt more correct to do the refactor instead of passing sections with rows to GridCollection. Still checking for any breaks, can always revert that commit if things go wrong since the commit before it resolves the weird `state.collection.getItem` (aka getting node via the collection's keymap)  vs `getChildNodes(node, state.collection)` (aka getting node via a node's `childNodes`) disparity